metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -988,52 +988,75 @@ class MetalsLanguageServer(
   }
 
   @JsonNotification("metals/didFocusTextDocument")
-  def didFocus(uri: String): CompletableFuture[DidFocusResult.Value] = {
-    val path = uri.toAbsolutePath
-    focusedDocument = Some(path)
-    buildTargets
-      .inverseSources(path)
-      .foreach(focusedDocumentBuildTarget.set)
+  def didFocus(
+      params: AnyRef
+  ): CompletableFuture[DidFocusResult.Value] = {
+
+    val uriOpt: Option[String] = params match {
+      case string: String =>
+        Option(string)
+      case (h: String) :: Nil =>
+        Option(h)
+      case _ =>
+        None
```

Comment:
Only suggestion is _maybe_ to log something out here if we actually hit this. Since like you said an `Object` is actually valid as a notification. Actually even more then that if this is `nil` by chance for some reason from the client it might be nice to have something in here like:
```suggestion
      scribe.warn(s"Unexpected notification params received for didFocusTextDocument: $parms")
```
wdyt?

Reply:
Added!